bindings/pydeck/pydeck/bindings/layer.py

```diff
@@ -127,6 +124,8 @@ def data(self, data_set):
         """
         if self.use_binary_transport:
             self._binary_data = self._prepare_binary_data(data_set)
+        elif is_geopandas_df(data_set) and self.type == "GeoJsonLayer":
```

Comment:
I also think this is potentially quite confusing because the behavior changes substantially depending on the type key.

Given that objects that have a `__geo_interface__` key aren't accepted in their object form, I'd call this regardless of the layer type

Reply:
I'm guessing that calling for all types, and showing errors for non-GeoJson types will be simpler/less surprising than only transforming for the GeoJsonLayer